src/librustc_typeck/check/op.rs

```diff
@@ -528,18 +528,18 @@ impl<'a, 'gcx, 'tcx> FnCtxt<'a, 'gcx, 'tcx> {
                    on the left and may require reallocation. This \
                    requires ownership of the string on the left";
         debug!("check_str_addition: {:?} + {:?}", lhs_ty, rhs_ty);
+        let is_std_string = |ty, s| &format!("{:?}", ty) == "std::string::String";
```

Comment:
And give some air to this closure (a newline before and after)...

You really like compact mode, eh..? =P